### PR TITLE
Add allowlist and blocklist feature

### PR DESCRIPTION
**Pull request by Aptcoder**
This PR introduces support for globally controlling where the extension should or shouldn’t parse using a combination of:
- A blocklist of domains or paths to exclude
- An allowlist of domains or paths to explicitly include
- A defaultAllowMode toggle that determines the default behavior

Behavior
When defaultAllowMode is true:
All sites are allowed unless they match an item in the blocklist.
Allowlist can override specific blocked paths.

When defaultAllowMode is false:
All sites are blocked unless they match an item in the allowlist.
Blocklist can override specific allowed paths.

Matching is done using string patterns that support full hostnames (e.g., example.com) and optional paths (e.g., example.com/sensitive).

Example
```ts
defaultAllowMode = true
blocklist = ['google.com']
allowlist = ['google.com/mail']
```
In this setup:
google.com is blocked
google.com/mail is explicitly allowed and will override the block

UI Changes
Adds editable lists for allowlist and blocklist in the options page
Includes UI toggle for defaultAllowMode to control the extension’s default behavior

Screenshot 2025-05-31 at 22 06 38

## Summary by CodeRabbit

* **New Features**
  * Added domain allowlist/blocklist feature to control where the extension operates
  * Users can specify domains using allow or block mode with wildcard support (e.g., *.example.com)
  * New settings interface for managing site filters
  * Color-coded status indicator showing whether name replacement is active or disabled on the current site
  * Extension icon now reflects the filtering status of the current site

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## v2.2.0
+
+- Added domain allowlist/blocklist feature to control where the extension operates
+  - Users can specify domains where the extension should or should not run
```
Comment (arimgibson) [inline CHANGELOG.md:6]: *based on allow or block mode

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## v2.2.0
+
+- Added domain allowlist/blocklist feature to control where the extension operates
```
Comment (arimgibson) [inline CHANGELOG.md:5]: If we can add something about the override feature that'd be huge!

**Review by arimgibson — CHANGES_REQUESTED** [review]
Really nice job on this overall @Aptcoder !! Super happy with this -- by far one of the most recommended features of the extension, and this is really solid.

First off, can you review all the commits I pushed and make sure they make sense? Don't want to have accidentally broken anything :) thanks!

Some quick feedback, accompanied by the video:
- Move Site Filtering to a service with super comprehensive unit tests
  - Ensure unit tests matching some of the edge cases I found in video; update algorithm to ensure all pass :)
- Update the TagInput to use [OnuUI's inputs](https://onu.zyyv.dev/components/input) 
- Enable backspace to remove records from the tags
- Auto-add tags when the user clicks "Save Settings"
- Update the parsing status to show which record in allowlist/blocklist (or defaultAllow) is causing the site to be displayed
- Consider using a pre-built match pattern library like [browser-extension-url-match](https://www.npmjs.com/package/browser-extension-url-match) (this is more of a spike task; overall I think you're probably right for having our own custom logic but thought this might be worth looking into)
- Use the `blocked` version of the extension icons on sites where the extension is blocked (I'll provide these later in the week, will try to get to ASAP!)
- Unit tests on the validateUrl portion as well since that's a custom regex (but agree with having a custom regex; Valibot's URL won't match in the way that we need without the protocol in front)

Video -- too big for GitHub to allow me to upload :( -- sorry I was so brief and short but really happy with this! Was just trying to make the file size small enough for GitHub to let me upload it lol
https://nx70024.your-storageshare.de/s/q7FoQaZYLN9aANC

**Review by arimgibson — APPROVED** [review]
@Aptcoder sorry for letting this sit for so long!! Made a few edits while trying to make sure I understand how it works; super happy with this 😊 the tests especially are fantastic

**Comment by arimgibson** [comment]
@coderabbit rereview

File: entrypoints/popup/App.svelte [hunk 3]
```diff
@@ -15,18 +15,25 @@
   import StealthMode from '@/components/StealthMode.svelte'
   import WarningIcon from '@/components/WarningIcon.svelte'
   import InfoIcon from '@/components/InfoIcon.svelte'
+  import type { ParsingStatus } from '@/utils/types'
+  import { SiteFiltering } from '@/services/siteFiltering'
+
```
[the remark this answers is not in the document]
Comment (arimgibson) [reply entrypoints/popup/App.svelte:20]: The first comment is not accurate, Svelte 5 moved away from the `on:` syntax instead to use properties like `onclick` (without the colon).

https://svelte.dev/docs/svelte/v5-migration-guide#Event-changes

File: utils/validations.ts [hunk 4]
```diff
@@ -48,3 +49,11 @@ export function validateNoRecursiveMappings(nameMappings: Names) {
...
   return !recursiveMappings
 }
+
+export const validURLMatcher = {
+  match: (url: string) => {
+    // Allow wildcard patterns and flexible domain/path matching for site filtering
+    // Supports: domain.com, *.domain.com, domain.com/path*, https://domain.com, etc.
+    return /^(https?:\/\/)?[\w*.-]+\.[\w*.-]+([/\w*._~:/?#[\]@!$&'()+,;=%-]*)?$/.test(url)
+  },
+}
```
[the remark this answers is not in the document]
Comment (arimgibson) [reply utils/validations.ts:59]: Ignoring, doesn't feel important right now. Can revisit if people have issues
